pkg/execution/executor/executor.go

```diff
@@ -829,31 +829,51 @@ func (e *executor) Resume(ctx context.Context, pause state.Pause, r execution.Re
 	return nil
 }
 
-func (e *executor) HandleGeneratorResponse(ctx context.Context, gen []*state.GeneratorOpcode, item queue.Item) error {
-	if len(gen) > 1 {
-		md, err := GetFunctionRunMetadata(ctx, e.sm, item.Identifier.RunID)
-		if err != nil || md == nil {
-			return fmt.Errorf("error loading function metadata: %w", err)
+func (e *executor) HandleGeneratorResponse(ctx context.Context, resp *state.DriverResponse, item queue.Item) error {
+	md, err := GetFunctionRunMetadata(ctx, e.sm, item.Identifier.RunID)
+	if err != nil || md == nil {
+		return fmt.Errorf("error loading function metadata: %w", err)
+	}
+
+	var update *state.MetadataUpdate
+	// NOTE: We only need to set hash versions when handling generator responses, else the
+	// fn is ending and it doesn't matter.
+	if md.RequestVersion == -1 {
+		update = &state.MetadataUpdate{
+			Context:                   md.Context,
+			Debugger:                  md.Debugger,
+			DisableImmediateExecution: md.DisableImmediateExecution,
+			RequestVersion:            resp.RequestVersion,
 		}
+	}
 
+	if len(resp.Generator) > 1 {
```

Comment:
What does `len(resp.Generator) > 1` mean? That at least 1 generator step was executed?

Reply:
Parallel execution, as there's more than one thing to run.